### PR TITLE
Fully normalize observables in `BaseEstimator.run()` method

### PR DESCRIPTION
**Pull request by pedrorrivero**
### Summary
Use `init_observable` logic from `primitives.utils` to fully normalize `observables` arg in `BaseEstimator.run()`. Normalization is now performed to the more specific type `tuple[SparsePauliOp, ...]`.

### Details and comments
As a bonus of this new normalization, `str` types are now accepted both standalone, or as elements in a `Sequence` object.

### CODE REVIEW

File: qiskit/primitives/base/base_estimator.py [hunk 1]
```diff
@@ -255,19 +255,15 @@ def _run(
     ################################################################################
     @staticmethod
     def _validate_observables(
-        observables: Sequence[BaseOperator | PauliSumOp] | BaseOperator | PauliSumOp,
-    ) -> tuple[BaseOperator | PauliSumOp, ...]:
-        if isinstance(observables, (BaseOperator, PauliSumOp)):
+        observables: Sequence[BaseOperator | PauliSumOp | str] | BaseOperator | PauliSumOp | str,
+    ) -> tuple[SparsePauliOp, ...]:
+        if not isinstance(observables, Sequence):
```
Comment (t-imamichi) [inline qiskit/primitives/base/base_estimator.py:260]: You may simplify the code by combining the condition before return.
```suggestion
        if isinstance(observables, str) or not isinstance(observables, Sequence):
```

**Comment by t-imamichi** [comment]
Thank you for simplifying the code and improving the test.
I think we need to carefully allow `str` as input carefully due to the following two reasons.

1. We cannot easily deprecate `str` as input in the future. I feel it's not so complicated to pass `Pauli(str)`.
2. The current implemenation of `init_observable` happens to handle `str` appropriately, but it's not intended (as the type hint implies) and not always well-tested. 
https://github.com/Qiskit/qiskit-terra/blob/8bfb058b9d94353936d7e2aea61015095487b873/qiskit/primitives/utils.py#L45

I would like to hear @ikkoham's impression.

**Comment by pedrorrivero** [comment]
Thanks for the review @t-imamichi 

I did not notice that `str` was not part of the `init_observable` signature; nonetheless, as I read the the function, I think it handles that case just right and it should be added. I will take care of it.

File: qiskit/primitives/base/base_estimator.py (hunk 1 above)
Comment (pedrorrivero) [reply qiskit/primitives/base/base_estimator.py]: Done! Thanks a lot! 🙂

**Comment by ikkoham** [comment]
Thank you. I don't want to be constrained by the type constraint here. I think we should document that it is passed to the `SparsePauliOp` constructor. SparsePauliOp specifications may be extended in the future.

**Review by ikkoham — COMMENTED** [review]
Normalization of `_run` method type signature looks good to me.

**Comment by pedrorrivero** [comment]
@t-imamichi 

> I personally don't want to allow str as input explicitly to keep the spec simple because third party also needs to follow BaseEstimator interfaces to implement their own implementation. 

Third parties need to implement `_run()`, not `run()`. So this PR actually makes it easier for them as they now only need to consider one type: `SparsePauliOp`.

> Why do we need to allow str explicitly?

It is the easiest way for users to provide an observable. It improves usability, with no extra cost for developers (i.e. since we are normalizing beforehand for them).

> Another point is that your implentation allows not only str but also list[str] as follows. Do you want to add list[str] as input too?

Yes! Just as we do with other types (e.g. `PauliSumOp`), we allow single input as well as Sequences for multiple inputs.

**Comment by pedrorrivero** [comment]
@ikkoham 

> Thank you. I don't want to be constrained by the type constraint here. I think we should document that it is passed to the SparsePauliOp constructor. SparsePauliOp specifications may be extended in the future.

Not sure I understand where you are going, could you elaborate? 🙂

**Comment by t-imamichi** [comment]
OK. I see the necessity of string as input. I ask you to add a mention of "String representation" of https://qiskit.org/documentation/stubs/qiskit.quantum_info.Pauli.html as the spec of string format.

**Comment by pedrorrivero** [comment]
Reference added @t-imamichi! That was a great idea, let me know if you like it or prefer a different format 😉

**Comment by ikkoham** [comment]
My point is that we can actually take a wider class here (`str` , `list[str]`, `PauliList`...). But to list them all would only make it harder, so I think it is better to just document that everything that can be passed to the PauliList constructor is possible. I don't recommend to write everything in type hints.
If PauliList is extended in the future, we will have to modify it as well.

File: qiskit/primitives/base/base_estimator.py [hunk 2]
```diff
@@ -203,7 +203,9 @@ def run(
 
         Args:
             circuits: one or more circuit objects.
-            observables: one or more observable objects.
+            observables: one or more observable objects. Several formats are allowed;
+                importantly, ``str`` should follow the string representation format for
+                :py:class:`~qiskit.quantum_info.Pauli` objects.
```
Comment (t-imamichi) [inline qiskit/primitives/base/base_estimator.py:208]:
```suggestion
                :class:`~qiskit.quantum_info.Pauli` objects.
```

**Comment by pedrorrivero** [comment]
@ikkoham that is actually a very fair point! I am not sure how to address it though...

The only thing that come to mind is define a specific type such as:
```python
OperatorLike = Union[...]

class SparsePauliOp:
  def __init__(operator: OperatorLike):
    ...

def init_observable(observable: OperatorLike):
  ...
```
which involves making changes in `quantum_info`. Maybe it is for the better. 
Let me know what you want to do!

**Comment by ikkoham** [comment]
Since `quantum_info` did not introduce type hints in the first place, it should be done when we introduce the type hint, but it is left as a future issue.

**Comment by ikkoham** [comment]
BTW, I note that we will deprecate `PauliSumOp` from this in the future since opflow will be deprecated.

**Comment by pedrorrivero** [comment]
Thanks @ikkoham @t-imamichi , is there anything else left to do in this PR?

**Review by t-imamichi — APPROVED** [review]
LGTM. Thank you for updating the PR.

**Review by ikkoham — APPROVED** [review]
Thanks. LGTM

**Pull request merged**